packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumPlugin.kt

```diff
@@ -275,6 +286,222 @@ class DatadogRumPlugin(
             )
         }
     }
+
+    @Suppress("TooGenericExceptionCaught")
+    internal fun <T> callEventMapper(
+        mapperName: String,
+        event: T,
+        encodedEvent: Map<String, Any?>,
+        completion: (Map<String, Any?>?, T) -> T?
+    ): T? {
+        var modifiedJson: Map<String, Any?>? = encodedEvent
+        val latch = CountDownLatch(1)
+
+        val handler = Handler(Looper.getMainLooper())
+        handler.post {
+            try {
+                channel.invokeMethod(
+                    mapperName,
+                    mapOf(
+                        "event" to encodedEvent
+                    ),
+                    object : Result {
+                        override fun success(result: Any?) {
+                            modifiedJson = (result as? Map<String, Any?>)
+                            latch.countDown()
+                        }
+
+                        override fun error(
+                            errorCode: String,
+                            errorMessage: String?,
+                            errorDetails: Any?
+                        ) {
+                            latch.countDown()
+                        }
+
+                        override fun notImplemented() {
+                            Datadog._internal._telemetry.error(
+                                "$mapperName returned notImplemented."
+                            )
+                            latch.countDown()
+                        }
+                    }
+                )
+            } catch (e: Exception) {
+                Datadog._internal._telemetry.error("Attempting call $mapperName failed.", e)
+                latch.countDown()
+            }
+        }
+
+        try {
+            // Stalls until the method channel finishes
+            if (!latch.await(1, TimeUnit.SECONDS)) {
+                Datadog._internal._telemetry.debug("$mapperName timed out")
+                return event
+            }
+
+            if (modifiedJson?.containsKey("_dd.mapper_error") == true) {
+                return event
+            }
+
+            return completion(modifiedJson, event)
+        } catch (e: InterruptedException) {
+            Datadog._internal._telemetry.debug(
+                "Latch await was interrupted. Returning unmodified event.",
+            )
+        } catch (e: Exception) {
+            Datadog._internal._telemetry.error(
+                "Unknown exception attempting to deserialize mapped log event." +
+                    " Returning unmodified event.",
+                e
+            )
+        }
+        
+        return event
+    }
+
+    internal fun mapViewEvent(event: ViewEvent): ViewEvent {
+        var jsonEvent = event.toJson().asMap()
+        jsonEvent = extractExtraUserInfo(jsonEvent)
```

Comment:
I think using `var` and then re-assigning it to the result of `extractExtraUserInfo` method can be misleading (I was thinking `extractUserInfo` method returns user info by looking on the name, but turns out it moves data inside the event).

Maybe something like `val jsonEvent = normalize(event.toJson().asMap())`?. `normalize` is not a good name obviously, we need to come up with something better, but this is the idea.

Reply:
Yeah, it's a good call, I had similar thoughts. `normalizeUserInfo` actually works as a name though. I'll change it.